File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min =Integer.MAX_VALUE;
        int maxProf = 0;
        for(int i=0;i<prices.length;i++){
            if(prices[i] < min){
                min = prices[i]; 
            }else if(prices[i]-min >maxProf){
                maxProf = prices[i]-min;
            }
        }
        return maxProf;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini = prices[0];
       // int maxi=prices[0];
        int ans=0;
        for(int price:prices){
            ans =max(ans,price-mini);
            mini=min(mini,price);
        }
        return ans;
    }
};